cmd/libs/go2idl/parser/parse.go

```diff
@@ -233,6 +234,22 @@ func (b *Builder) AddDirRecursive(dir string) error {
 	return nil
 }
 
+// AddDirTo adds an entire directory to a given Universe. Unlike AddDir, this
+// processes the package immediately, which makes it safe to use from withing a
+// generator (rather than just at init time. 'dir' must be a single go package.
+// GOPATH, GOROOT, and the location of your go binary (`which go`) will all be
+// searched if dir doesn't literally resolve.
+func (b *Builder) AddDirTo(dir string, u *types.Universe) error {
+	if _, found := b.parsed[dir]; !found {
+		if err := b.addDir(dir, true); err != nil {
+			return err
+		}
+	} else {
+		b.userRequested[dir] = true
```

Comment:
I don't understand the logic behind setting userRequested in else here.
Can you please clarify (by adding a comment?)